test/e2e/clusterset_test.go

```diff
@@ -0,0 +1,125 @@
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+			// create clusterRoleBinding to real cluster
+			clusterrolebinding, err = util.CreateResource(dynamicClient, clusterRoleBindingGVR, clusterrolebinding)
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred(), "Failed to create %s", clusterRoleBindingGVR.Resource)
+		})
+
+		ginkgo.It("clusterrolebinding should be auto created successfully", func() {
+			gomega.Eventually(func() (interface{}, error) {
+				clusterroleBindingName := util.GenerateClusterRoleBindingName("cluster1")
+				return util.HasClusterResource(dynamicClient, clusterRoleBindingGVR, clusterroleBindingName)
+			}, eventuallyTimeout, eventuallyInterval).Should(gomega.BeTrue())
+		})
+
+		ginkgo.It("update clusterrolebinding", func() {
+			clusterrolebinding, err = util.LoadResourceFromJSON(util.ClusterRoleBindingTemplate)
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			err = util.SetSubjects(clusterrolebinding, updatedSubject)
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+			clusterrolebinding, err = util.UpdateClusterResource(dynamicClient, clusterRoleBindingGVR, clusterrolebinding)
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		})
+
+		ginkgo.It("clusterrolebinding should be auto updated successfully", func() {
+			gomega.Eventually(func() (interface{}, error) {
+				clusterroleBindingName := util.GenerateClusterRoleBindingName("cluster1")
+				generatedClusterrolebinding, err := util.GetClusterResource(dynamicClient, clusterRoleBindingGVR, clusterroleBindingName)
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+				subjects, _, err := unstructured.NestedSlice(generatedClusterrolebinding.Object, "subjects")
+				gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+
+				for _, subject := range subjects {
+					subjectValue, _ := subject.(map[string]interface{})
+					if subjectValue["kind"] == updatedSubject.Kind &&
+						subjectValue["name"] == updatedSubject.Name &&
+						subjectValue["namespace"] == updatedSubject.Namespace {
+						return true, nil
+					}
+				}
+				return false, nil
+			}, eventuallyTimeout, eventuallyInterval).Should(gomega.BeTrue())
+		})
+
+		ginkgo.It("should delete successfully", func() {
+			err := util.DeleteClusterResource(dynamicClient, clusterSetGVR, clusterset.GetName())
+			gomega.Expect(err).ShouldNot(gomega.HaveOccurred())
+		})
+		ginkgo.It("clusterrolebinding should be auto deleted successfully", func() {
+			gomega.Eventually(func() (interface{}, error) {
+				clusterroleBindingName := util.GenerateClusterRoleBindingName("cluster1")
+				return util.HasClusterResource(dynamicClient, clusterRoleBindingGVR, clusterroleBindingName)
+			}, eventuallyTimeout, eventuallyInterval).Should(gomega.BeFalse())
+		})
+		ginkgo.It("delete clusterrole/binding successfully", func() {
+			err = util.DeleteClusterResource(dynamicClient, clusterRoleGVR, clusterrole.GetName())
```

Comment:
i am not sure the intent here. clean up?

Reply:
yes, clean up

Reply:
how about using aftereach?

Reply:
The clusterrole and clusterrolebinding is created in "ginkgo.It", and I think it's not need to create/update them in before each and aftereach.  It's just need to create once. what's your opinion?

Reply:
I think having aftereach to clean all the test data seems reasonable. You could use label to delete all the test data. So that would also ease your  life when you have another case, such as "updating clusterset".